test/erc-20-extensions/ERC20Extensions.js

```diff
@@ -252,3 +254,49 @@ describe('ERC20ExtensionsMock tests', function () {
     })
   })
 })
+
+// Transaction needs to be propagated to the mirror node
+async function pollForChangedSupply(ERC20Burnable, initialSupply){
+  let numberOfTries = 0;
+  const timesToTry = 300;
+  let newSupply = 0;
+
+  while (numberOfTries < timesToTry) {
+
+    newSupply = await ERC20Burnable.totalSupply()
+
+
+    if ((newSupply != 0) && (newSupply != initialSupply)) {
+      return newSupply;
+    }
+
+    numberOfTries++;
+    await delay(4000); // Delay for 4 seconds before the next attempt
+
+    throw new Error(`ERC20Burnable failed to change after ${timesToTry} tries`);
+  }
+}
+
+// Transaction needs to be propagated to the mirror node
+async function unPauseAndPoll(ERC20Pausable) {
+  let numberOfTries = 0;
+  const timesToTry = 400;
+
+  await ERC20Pausable.unpause()
+  while (numberOfTries < timesToTry) {
+    const paused = await ERC20Pausable.paused()
+    console.log("paused: ", paused)
+    if (!await ERC20Pausable.paused()) {
+      return false;
+    }
+
+    numberOfTries++;
+    await delay(5000); // Delay for 5 seconds before the next attempt
```

Comment:
You delay 5s here and 4s before.
Aren't both cases working around the E2E transaction time?
Feel like these should be a default class constant of delay time. Maybe even configurable that tests use

Reply:
The polling has been consolidated into the helper.js file and they all use the configurable RETRY_DELAY environment variable.